### ISSUE TITLE
False positive "Cannot borrow immutable as mutable "

### ISSUE THREAD

**Issue by MatrixDev** [intellij-rust/intellij-rust#9799]
## Environment

* **IntelliJ Rust plugin version:** 0.4.184.5027-223-nightly
* **Rust toolchain version:** rustc 1.65.0 (897e37553 2022-11-02)
* **IDE name and version:** CLion Build #CL-223.7401.10, built on November 2, 2022
* **Operating system:** MacOS Monterey 12.6 (arm64)

## Problem description
IDE displays an error while code compiles successfully.
Happens when RefCell contains a mutable reference and we access it without storing RefMut into the variable.

## Steps to reproduce
This is the most simplified example as I could come up with:
```rust
let mut option = None;
let cell = RefCell::new(&mut option);
cell.borrow_mut().insert(0);
```

Error:
```Cannot borrow immutable local variable `cell.borrow_mut()` as mutable```

<img width="569" alt="image" src="https://user-images.githubusercontent.com/1215187/203976684-23ccaf9a-a24a-43e0-8e33-b07692e392ad.png">
